src/components/TodoContainer.js

```diff
@@ -0,0 +1,216 @@
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+      body: JSON.stringify(airtableData),
+    };
+
+    try {
+      // Send a POST request to add a new todo. Fetch data from the API
+      const response = await fetch(`${AirtableUrl}`, options);
+
+      // Check if the response is not successful
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const addedTodo = await response.json();
+      setTodoList((prevTodoList) => [
+        ...prevTodoList,
+        { id: addedTodo.id, title: addedTodo.fields.title },
+      ]);
+      //  update the local state directly
+
+      // Handle errors during adding a todo
+    } catch (error) {
+      console.log("Error fetching data:", error.message);
+      return null;
+    }
+  };
+
+  //Delete items method DELETE
+  const removeTodo = async (id) => {
+    const options = {
+      method: "DELETE",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+    };
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        throw new Error(`Error has occurred: ${response.status}`);
+      }
+
+      fetchData(); //deleting items from todo list
+
+      return await response.json();
+    } catch (error) {
+      console.log(error.message);
+      return null;
+    }
+  };
+
+  //Checkbox  Method PATCH
+  const updateData = useCallback(async (id, newValue) => {
+    const options = {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        // "Authorization": `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`
+      },
+      body: JSON.stringify({
+        fields: {
+          CheckBox: newValue.CheckBox,
+        },
+      }),
+    };
+
+    console.log(`Updating record with id: ${id}, newValue: ${newValue}`);
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+      // console.log(data); // Log the response data to the console
+      // console.log(`Updating record with id: ${id}, newValue:`, newValue);
+      return data;
+    } catch (error) {
+      console.log("Error updating data: ", error);
+    }
+  }, []);
+
+  // Sorting by JavaScript
+  //   The sorting order is determined by the sortOrder state variable.
+  //   After sorting, the sorted list is set as the new state for todoList.
+  //
+
+  const sortData = useCallback(() => {
+    setTodoList((prevTodoList) => {
+      const sorted = [...prevTodoList].sort((objectA, objectB) => {
```

Comment:
Nitpick:
Instead of using the spread operator, there's an [Array.from()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) method that creates a new array from another array.